tests/unit_tests/plugins/modules/readers/test_readers.py

```diff
@@ -0,0 +1,88 @@
+# # # Distribution Statement A. Approved for public release. Distribution unlimited.
+# # #
+# # # Author:
+# # # Naval Research Laboratory, Marine Meteorology Division
+# # #
+# # # This program is free software: you can redistribute it and/or modify it under
+# # # the terms of the NRLMMD License included with this program. This program is
+# # # distributed WITHOUT ANY WARRANTY; without even the implied warranty of
+# # # MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the included license
+# # # for more details. If you did not receive the license, for more information see:
+# # # https://github.com/U-S-NRL-Marine-Meteorology-Division/
+
+"""Unit tests on all the readers."""
+import pytest
+from geoips.commandline.log_setup import setup_logging
+from geoips.interfaces import readers
+
+
+LOG = setup_logging()
+
+
+class TestReaders:
+    """Unit tests every reader."""
+
+    available_readers = [
+        ("abi_l2_netcdf"),
+        ("abi_netcdf"),
+        ("ahi_hsd"),
+        ("amsr2_netcdf"),
+        ("amsr2_remss_winds_netcdf"),
+        ("amsub_hdf"),
+        ("amsub_mirs"),
+        ("ascat_uhr_netcdf"),
+        ("atms_hdf5"),
+        ("ewsg_netcdf"),
+        ("geoips_netcdf"),
+        ("gmi_hdf5"),
+        ("imerg_hdf5"),
+        ("mimic_netcdf"),
+        ("modis_hdf4"),
+        ("saphir_hdf5"),
+        ("sar_winds_netcdf"),
+        ("scat_knmi_winds_netcdf"),
+        ("scat_noaa_winds_netcdf"),
+        ("seviri_hrit"),
+        ("sfc_winds_text"),
+        ("smap_remss_winds_netcdf"),
+        ("smos_winds_netcdf"),
+        ("ssmi_binary"),
+        ("ssmis_binary"),
+        ("viirs_netcdf"),
+        ("wfabba_ascii"),
+        ("windsat_idr37_binary"),
+        ("windsat_remss_winds_netcdf"),
+    ]
```

Comment:
@jsolbrig @AJThorpe  Should we use entry points here to identify all the available readers, rather than explicitly listing them?  I am guessing yes?

Reply:
@mindyls Added entry point registration, similar to Evans method, works well!